components/prism-nginx.js

```diff
@@ -0,0 +1,18 @@
+
+Prism.languages.nginx = Prism.languages.extend('clike', {
+        comment: {
+                pattern: /(^|[^"{\\])(#.*?(\r?\n|$))/,
```

Comment:
I think you'll need a `lookbehind: true` here, so that no char is consumed before the comment.
Also, you probably don't need the last group ; `/(^|[^"{\\])#.*/` should be enough (since it does not match line feeds).